src/pybind/mgr/cephadm/templates/services/grafana/ceph-dashboard.yml.j2

```diff
@@ -16,3 +19,12 @@ datasources:
     isDefault: {{ 'true' if loop.first else 'false' }}
     editable: false
 {% endfor %}
+
+  - name: 'Loki'
+    type: 'loki'
+    access: 'proxy'
+    orgId: 1
+    url: '{{ loki_host }}'
+    basicAuth: false
+    isDefault: true
+    editable: false
```

Comment:
I'm comparing this to the Prometheus service (above) and I'm wondering... What's the high-availability model for Loki+Promtail? If I want to ensure that my logs survive the loss of 1 node (let's say the one where Loki is), what are the deployment guidelines for that scenario? In the case of Prometheus it's merely deploying 2, 3 instances.

Reply:
We need to redeploy the grafana service in this case. The next host in the list will be configured in the Loki datasource instead of the one that crashed.